src/main/java/com/nvidia/spark/rapids/jni/DecimalUtils.java

```diff
@@ -55,9 +55,23 @@ public static Table multiply128(ColumnView a, ColumnView b, int productScale) {
    *         row.
    */
   public static Table divide128(ColumnView a, ColumnView b, int quotientScale) {
-    return new Table(divide128(a.getNativeView(), b.getNativeView(), quotientScale));
+    return new Table(divide128(a.getNativeView(), b.getNativeView(), quotientScale, false));
   }
 
+  /**
+   * Divide two DECIMAL128 columns and produce a INT64 quotient with overflow detection.
+   * This method considers a precision greater than 19 as overflow even if the number still fits in
```

Comment:
There are a few things wrong with this.  A Decimal(19,0) can hold all INT64 values plus some.  When I first suggested that we should do a regular divide and set the result to Decimal(19,0) and then cast it to a long I thought that the divide used in integral divide would do rounding. But it does not. We want to be sure that the output fits in 64 bits, not Decimal(19,0).

Reply:
I will cast the output to INT64 as per docs. It was late last night when I pushed, I wanted people to see the approach before I did a cleanup. Thanks for the feedback